src/components/ProblemsTable.tsx

```diff
@@ -266,36 +353,64 @@ export const ProblemsTable = ({
     [context, displayStrings, filetypeIcons, getFileNameFromId, sortByLevel]
   );
 
+  const reorderColumn = React.useCallback(
+    (column: Column<TableRow>[]) => {
+      const currentTable = context?.currentTable;
+      let toplevel = currentTable ? (tableStyleAccessor[currentTable] as keyof Report) : undefined;
+
+      if (!toplevel || currentTable === 'problems') {
+        return column;
+      }
+
+      if (toplevel === 'fileId') {
+        toplevel = 'fileName';
+      }
+
+      const topPosition = column.findIndex((col) => col.id === toplevel);
```

Comment:
there are no checks if topPosition is not found by index, later it is used in splice and in unshif we say that topCol always has a [0]

Reply:
There is a check on line 361 for a defined topLevel, which must be either 'fileId' or 'category', (which is a subsequent check of defined context.currentTable) - otherwise original column argument is returned. The column argument will [currently] always have a column whose key will match one of those two possibilities in topLevel